src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.String.cs

```diff
@@ -220,47 +322,76 @@ public void WriteStringValue(ReadOnlySpan<byte> utf8Value)
         {
             JsonWriterHelper.ValidateValue(utf8Value);
 
-            WriteStringEscape(utf8Value);
+            WriteStringEscape(utf8Value, JsonTokenType.String);
 
             SetFlagToAddListSeparatorBeforeNextItem();
             _tokenType = JsonTokenType.String;
         }
 
-        private void WriteStringEscape(ReadOnlySpan<byte> utf8Value)
+        /// <summary>
+        /// Writes the UTF-8 text value segment as a partial JSON string.
+        /// </summary>
+        /// <param name="utf8Value">The UTF-8 encoded value to be written as a JSON string element of a JSON array.</param>
+        /// <param name="isFinalSegment">Indicates that this is the final segment of the string.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified value is too large.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this would result in invalid JSON being written (while validation is enabled).
+        /// </exception>
+        /// <remarks>
+        /// The value is escaped before writing.
+        /// </remarks>
+        public void WriteStringValueSegment(ReadOnlySpan<byte> utf8Value, bool isFinalSegment)
+        {
+            JsonWriterHelper.ValidateValue(utf8Value);
+
+            JsonTokenType nextTokenType = isFinalSegment ? JsonTokenType.String : JsonTokenType.StringSegment;
+            WriteStringEscape(utf8Value, nextTokenType);
```

Comment:
This needs to account for UTF-8 code points whose bytes are split between two segments, as well as surrogate pairs like with `ReadOnlySpan<char>`